partner_create_by_vat/models/res_partner.py

```diff
@@ -0,0 +1,160 @@
+        elif part.name and part.name != " " and _check_vat_ro(part.name):
+            self.write({'vat': part.name.upper().replace(" ", "")})
+        if not part.vat and part.name:
+            raise except_orm(_('Error!'), _("No VAT number found"))
+
+        vat_country, vat_number = self._split_vat(part.vat)
+
+        if part.vat_subjected:
+            self.write({'vat_subjected': False})
+        if vat_number and vat_country:
+            self.write({
+                'is_company': True,
+                'country_id': self.env['res.country'].search(
+                    [('code', 'ilike', vat_country)])[0].id
+            })
+            nrc_key = 'Numar de inmatriculare la Registrul Comertului:'
+            tva_key = 'Taxa pe valoarea adaugata (data luarii in evidenta):'
+            if vat_country == 'ro':
+                try:
+                    result = getMfinante(vat_number)
+                    name = nrc = adresa = tel = fax = False
+                    zip1 = vat_s = state = False
+                    if 'Denumire platitor:' in result.keys():
+                        name = result['Denumire platitor:'].upper()
+                    if 'Adresa:' in result.keys():
+                        adresa = result['Adresa:'].title() or ''
+                    if nrc_key in result.keys():
+                        nrc = result[nrc_key].replace(' ', '')
+                        if nrc == '-/-/-':
+                            nrc = ''
+                    if 'Codul postal:' in result.keys():
+                        zip1 = result['Codul postal:'] or ''
+                    if 'Judetul:' in result.keys():
+                        jud = result['Judetul:'].title() or ''
+                        if jud.lower().startswith('municip'):
+                            jud = ' '.join(jud.split(' ')[1:])
+                        if jud != '':
+                            state = self.env['res.country.state'].search(
+                                [('name', 'ilike', jud)])
+                            if state:
+                                state = state[0].id
+                    if 'Telefon:' in result.keys():
+                        tel = result['Telefon:'].replace('.', '') or ''
+                    if 'Fax:' in result.keys():
+                        fax = result['Fax:'].replace('.', '') or ''
+                    if tva_key in result.keys():
+                        vat_s = bool(
+                            result[tva_key] != 'NU')
+                    self.write({
+                        'name': name or '',
+                        'nrc': nrc or '',
+                        'street': adresa or '',
+                        'phone': tel or '',
+                        'fax': fax or '',
+                        'zip': zip1 or '',
+                        'vat_subjected': vat_s or False,
+                        'state_id': state,
+                    })
+                except:
```

Comment:
It is usually bad practice to catch any exception instead of a specific one, especially on such a large block of code.